File: duplicates-in-string/duplicatesInString.js
function printDuplicates(sampleString) {
    let stringLength = sampleString.length;

    // Sorting the string
    sampleString = sampleString.split('').sort().join('');

    // Loop through the sorted string to find duplicates
    for (let i = 0; i < stringLength; i++) {
        let count = 1;

        // Counting the occurrences of each character
        while (i < stringLength - 1 && sampleString[i] == sampleString[i + 1]) {
            count++;
            i++;
        }

        // Printing the duplicate character and its count
        if (count > 1) {
            console.log(sampleString[i] + ", count = " + count);
        }
    }
}

let sampleString = "lee spangles";
printDuplicates(sampleString);

File: duplicates-in-string/duplicatesInString.java
import java.util.Arrays;

public class DuplicatesInString {

    public static void printDuplicates(String str) {
        int stringLength = str.length();

        // Sorting the string
        char[] chars = str.toCharArray();
        Arrays.sort(chars);
        String sortedStr = new String(chars);

        // Loop through the sorted string to find duplicates
        for (int i = 0; i < stringLength; i++) {
            int count = 1;

            // Counting the occurrences of each character
            while (i < stringLength - 1 && sortedStr.charAt(i) == sortedStr.charAt(i + 1)) {
                count++;
                i++;
            }

            // Printing the duplicate character and its count
            if (count > 1) {
                System.out.println(sortedStr.charAt(i) + ", count = " + count);
            }
        }
    }

    public static void main(String[] args) {
        String str = "lee spangles";
        printDuplicates(str);
    }
}
